PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/ServiceBusReceiverAsyncClientCheck.java

```diff
@@ -22,30 +16,28 @@
  */
 public class ServiceBusReceiverAsyncClientCheck extends LocalInspectionTool {
 
-    private static List<String> CLIENT_DATA = null;
-
     // Define constants for string literals
-    private static final String RULE_CONFIGURATION = "META-INF/ruleConfigs.json";
-    private static final String RULE_NAME = "ServiceBusReceiverAsyncClientCheck";
-    private static final String SUGGESTION_KEY = "antipattern_message";
-    private static final String CLIENT_NAME_KEY = "client_name";
-    private static final Logger LOGGER = Logger.getLogger(ServiceBusReceiverAsyncClientCheck.class.getName());
+    private static final String CLIENT_NAME;
+    private static final String SUGGESTION;
 
+    // Static initializer block to load the client data once
     static {
-        try {
-            CLIENT_DATA = getClientToCheck();
-        } catch (IOException e) {
-            LOGGER.log(Level.SEVERE, "Error loading client data", e);
-        }
-    }
+        final String ruleName = "ServiceBusReceiverAsyncClientCheck";
 
+        RuleConfigLoader centralRuleConfigLoader = RuleConfigLoader.getInstance();
+        final RuleConfig ruleConfig = centralRuleConfigLoader.getRuleConfig(ruleName);
```

Comment:
It's possible that `getRuleConfig(ruleName)` returns null. If you look at the implementation of getRuleConfig, and the documentation for the `get` call in `list.get(String key)`, it will return null if the key does not exist.

In that case, you'll get a NullPointerException when trying to set CLIENT_NAME and SUGGESTION.

Reply:
discussed adding an empty class offline